### PR TITLE
Automatically fix any code style issues

### PR DESCRIPTION
**Pull request by frankieroberto**
This moves the code style tests into a separate GitHub workflow, where it will automatically fix any correctable issues and commit and push them, and then re-run the code style tests to see if there are any remaining issues.

### CODE REVIEW

**Comment by frankieroberto** [comment]
Hmm, this seems to commit all the gems within a `vendor/` folder. Need to avoid that...

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: 2.7.0
+        bundler-cache: true
+    - name: Run linters with autofix
+      run: bundle exec rubocop --auto-correct-all
+    - name: Push any fixes
+      run: |
+        git config --global user.email "lint-bot@example.com"
```
Comment (alaric-dfe) [inline .github/workflows/lint.yml:21]: We should probably not leave an `@example.com` address in there, but it doesn't have to be a real email address either. How about just `git config --global user-email "lint-bot"`? I tried an experiment and git doesn't seem to check it's a valid email address.

**Review by alaric-dfe — APPROVED** [review]
I like it

File: .github/workflows/lint.yml (hunk 1 above)
Comment (frankieroberto) [reply .github/workflows/lint.yml:21]: Good idea!

Comment (frankieroberto) [inline .github/workflows/lint.yml:21]:
```suggestion
        git config --global user.email "lint-bot"
```

**Pull request merged**